scripts/generate_cloudevent_protos.dart

```diff
@@ -26,55 +27,198 @@ void setupLogging() {
   // Ensure to change back to the previous directory securely after operation
-  final currentDirectory = Directory.current.path;
-  try {
-    Directory.current = 'cloudevents';
-    await Process.run('sh', ['scripts/compile.sh']);
-  } finally {
-    // Safely returning to the original directory even if the operation fails
-    Directory.current = currentDirectory;
+  await Process.run(
+    'sh',
+    ['scripts/compile.sh'],
+    workingDirectory: 'cloudevents',
+  );
+
+  // These paths are relative to the 'cloudevents' directory,
+  // which is the current working directory for protoc.
+  const outputPath = './lib/src';
+  final outputDir = Directory(outputPath);
+
+// Clean the outputPath directory if it exists to ensure a clean state for generation.
+// which is the current working directory for protoc.
+  if (!await _ensureCleanOutputDir(outputDir)) {
+    return false;
   }
 
   const protosPath = './cloudevents';
-  const outputPath = './lib/src';
-  final protocCommand = [
-    'protoc',
-    '-I$protosPath/proto',
-    '-I$protosPath/third_party/googleapis',
-    '-I$protosPath/tmp/include',
-    '--dart_out=$outputPath',
+  final protocArgsForFile = <String>[
+    '-I${_toProtocArgPath(path_lib.join(protosPath, 'proto'))}',
+    '-I${_toProtocArgPath(path_lib.join(protosPath, 'third_party', 'googleapis'))}',
+    '-I${_toProtocArgPath(path_lib.join(protosPath, 'tmp', 'include'))}',
+    '--dart_out=${_toProtocArgPath(outputPath)}',
   ];
 
-  // Dynamically add .proto files to the protoc command
+  // Dynamically add .proto files to the arguments list
   final directory = Directory(protosPath);
   if (directory.existsSync()) {
     await for (final entity
         in directory.list(recursive: true, followLinks: false)) {
       if (entity.path.endsWith('.proto')) {
-        protocCommand.add(entity.path);
+        // entity.path is relative to the CWD ('cloudevents')
+        // e.g., 'cloudevents/proto/google/events/cloud/audit/v1/data.proto'
+        protocArgsForFile.add(_toProtocArgPath(entity.path));
       }
     }
+  } else {
+    logger.warning('"$protosPath" not found; no .proto files to compile.');
+    return false;
   }
 
-  final result =
-      await Process.run(protocCommand.first, protocCommand.sublist(1));
+  ProcessResult? result;
 
-  // Move the generated files up a directory
-  await Process.run(
-    'bash',
-    ['-c', 'mv lib/src/google/* lib/src/ && rmdir lib/src/google'],
-  );
+  // To avoid issues with command line length limits (especially on Windows,
+  // but potentially on other OSes with a very large number of .proto files),
+  // we write all protoc arguments to a temporary file.
+  // Protoc can then read these arguments using the '@<filepath>' syntax.
+  // The following block handles the creation, usage, and cleanup of this temporary file.
+  Directory? tempDir;
+  File? argFile;
+
+  try {
+    tempDir = Directory.systemTemp.createTempSync('protoc_args_');
```

Comment:
Looks like this could throw either an [ArgumentError](https://github.com/dart-lang/sdk/blob/main/sdk/lib/io/directory_impl.dart#L183-L188) or a [FileSystemException](https://github.com/dart-lang/sdk/blob/main/sdk/lib/io/directory_impl.dart#L202-L207). I think we may want to follow [Bob's advice](https://stackoverflow.com/questions/17315945/error-vs-exception-in-dart) and only catch the FileSystemException.